zerver/migrations/0497_remove_prefix_from_archived_streams.py

```diff
@@ -0,0 +1,40 @@
+import hashlib
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def remove_prefix_from_archived_streams(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Stream = apps.get_model("zerver", "Stream")
+    archived_streams = Stream.objects.filter(deactivated=True)
+
+    for archived_stream in archived_streams:
+        streamID = str(archived_stream.id)
+        stream_id_hash_object = hashlib.sha512(streamID.encode())
+        hashed_stream_id = stream_id_hash_object.hexdigest()[0:7]
+        prefix = hashed_stream_id + "!DEACTIVATED:"
+        prefix_length = len(prefix)
+        old_name = archived_stream.name
+        if old_name.startswith(prefix):
```

Comment:
We should extend this migration to handful streams deactivated before 1b6f68bb59dc85125ff21fbd3f98a88f39c0730d.

Reply:
Done.